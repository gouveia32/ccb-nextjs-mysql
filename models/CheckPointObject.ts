export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }