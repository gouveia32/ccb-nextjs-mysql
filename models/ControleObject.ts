export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
