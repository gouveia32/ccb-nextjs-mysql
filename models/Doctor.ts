export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
