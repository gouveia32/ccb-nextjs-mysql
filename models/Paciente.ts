export interface PacienteModel {
  id: string;
  nome: string;
}

export type TipoPaciente = PacienteModel;

export const cPacienteModel = {
  id: "id",
  nome: "nome"
};

export const PacienteObject: TipoPaciente = {
  id: "",
  nome: "",
};

export function isPacienteType(object: any): object is TipoPaciente {
  return "nome" in object;
}
