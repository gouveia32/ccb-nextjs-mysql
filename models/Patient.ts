export interface PatientModel {
  id: string;
  name: string;
}

export type PatientType = PatientModel;

export const cPatientModel = {
  id: "id",
  name: "name"
};

export const PatientObject: PatientType = {
  id: "",
  name: "",
};

export function isPatientType(object: any): object is PatientType {
  return "name" in object;
}
