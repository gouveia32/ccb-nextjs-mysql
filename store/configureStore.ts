import {
  combineReducers,
  configureStore,
  getDefaultMiddleware,
} from "@reduxjs/toolkit";
import rootSaga, { mainReducers } from "./RootState";
import createSagaMiddleware from "redux-saga";

let store: any;

function configureAppStore() {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with saga middleware and router middleware
  const middlewares = [sagaMiddleware];

  const store = configureStore({
    reducer: combineReducers({ ...mainReducers }),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools: process.env.NODE_ENV !== "production",
  });

  // Add all combined Sagas to middleware
  sagaMiddleware.run(rootSaga);

  return store;
}

export const initializeStore = () => {
  let _store = store ?? configureAppStore();
  // For SSG and SSR always create a new store
  if (typeof window === "undefined") return _store;
  // Create the store once in the client
  if (!store) store = _store;

  return _store;
};
