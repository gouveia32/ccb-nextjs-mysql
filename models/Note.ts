import { TagType } from "./Tag";
import { CheckPointType } from "./CheckPointObject";
import { Note } from "@prisma/client";

export enum NoteTypeEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NoteModel {
  id?: string;
  name: string;
  noteType: NoteTypeEnum;
  content: string;
  color: string;
  pinned: boolean;
  createdAt: string;
  tags: TagType[];
  checkPoints?: CheckPointType[];
}

export type NoteType = NoteModel;

export const NoteObject: NoteType = {
  id: "",
  name: "",
  noteType: NoteTypeEnum.TEXT,
  content: "",
  color: "#fff",
  pinned: false,
  createdAt: new Date(Date.now()).toString(),
  tags: [],
  checkPoints: [],
};

export const NoteDBObject: Note = {
  id: "",
  name: "",
  noteType: "TEXT",
  color: "",
  content: "",
  pinned: false,
  createdAt: new Date(Date.now()),
  doctorId: "",
};

export enum cNoteModel {
  id = "id",
  name = "name",
  noteType = "noteType",
  color = "color",
  content = "content",
  pinned = "pinned",
  createdAt = "createdAt",
  tags = "tags",
  checkPoints = "checkPoints",
}
