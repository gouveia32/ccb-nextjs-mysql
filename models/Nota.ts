import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
