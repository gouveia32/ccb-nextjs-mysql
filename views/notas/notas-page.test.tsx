import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import NotasPage, { NotasPageProps } from "../../pages/notas/[index]";
import { NotaObject } from "../../models/Nota";
import { initializeStore } from "../../store/configureStore";
import { Provider } from "react-redux";

describe("Notas page", () => {
  let wrapper: ShallowWrapper;
  const store = initializeStore();

  beforeEach(() => {
    const mockProps: NotasPageProps = {
      session: null,
      userNotas: [NotaObject, NotaObject],
    };

    wrapper = shallow(
      <Provider store={store}>
        <NotasPage {...mockProps} />
      </Provider>
    );
  });

  it("should render Notas page", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
