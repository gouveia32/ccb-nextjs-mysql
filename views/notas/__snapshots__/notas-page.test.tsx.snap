// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notas page should render Notas page 1`] = `ShallowWrapper {}`;
